File: Binary Trees/invertBinaryTree.js
// recursive solution
const invertBinaryTree = (tree) => {
    return invertHelper(tree)
}
  
const invertHelper = (node) => {
    if (!node) return
    swapNodes(node)
    if (node.left) invertHelper(node.left)
    if (node.right) invertHelper(node.right)
    
    return node
}

//iterative solution
const invertBinaryTree1 = (tree) => {
    const queue = [tree]
    if (!tree) return
    while (queue.length) {
        const current = queue.shift()
        if (current === null) continue
        swapNodes(current)
        queue.push(current.left)
        queue.push(current.right)
    }
    return tree
}

const swapNodes = (node) => {
    [node.left, node.right] = [node.right, node.left]
}

File: Binary Trees/invert_binary_tree.py
# recursive solution
def invert_binary_tree(tree):
    return invert_helper(tree)

def invert_helper(node):
    if not node:
        return
    swapNodes(node)
    if node.left:
        invert_helper(node.left)
    if node.right:
        invert_helper(node.right)
    return node

#iterative solution
def invert_binary_tree1(tree):
    queue = [tree]
    while len(queue):
        current = queue.pop(0)
        if current is None:
            continue
        swapNodes(current)
        queue.append(current.left)
        queue.append(current.right)

def swapNodes(node):
	node.left, node.right = node.right, node.left